app/controllers/repositories_controller.rb

```diff
@@ -2,8 +2,13 @@ class RepositoriesController < SessionsController
   before_action :current_user
   before_action :signed_in, except: [:index, :show, :download, :download_files]
   before_action :set_repository, only: [:show, :add_like, :destroy, :edit, :update, :download_files]
+  before_action :check_session
+  before_action :check_auth
```

Comment:
`before_action` takes a list of actions that looks a bit cleaner. eg: `before_action :current_user, :check_auth, check_session`